pkg/reconciler/revision/revision.go

```diff
@@ -78,21 +93,76 @@ func (c *Reconciler) reconcileDigest(ctx context.Context, rev *v1.Revision) erro
 		ServiceAccountName: rev.Spec.ServiceAccountName,
 		ImagePullSecrets:   imagePullSecrets,
 	}
-	digest, err := c.resolver.Resolve(rev.Spec.GetContainer().Image,
-		opt, cfgs.Deployment.RegistriesSkippingTagResolving)
-	if err != nil {
-		err = fmt.Errorf("failed to resolve image to digest: %w", err)
-		rev.Status.MarkContainerHealthyFalse(v1.ReasonContainerMissing,
-			v1.RevisionContainerMissingMessage(
-				rev.Spec.GetContainer().Image, err.Error()))
-		return err
-	}
 
-	rev.Status.ImageDigest = digest
+	var digestGrp errgroup.Group
+	type digestData struct {
+		digestValue        string
+		containerName      string
+		isServingContainer bool
+		image              string
+		digestError        error
+	}
 
+	digests := make(chan digestData, len(rev.Spec.Containers))
+	for _, container := range rev.Spec.Containers {
+		container := container // Standard Go concurrency pattern.
+		digestGrp.Go(func() error {
+			digest, err := c.resolver.Resolve(container.Image,
+				opt, cfgs.Deployment.RegistriesSkippingTagResolving)
+			if err != nil {
+				err = fmt.Errorf("failed to resolve image to digest: %w", err)
+				digests <- digestData{
+					image:       container.Image,
+					digestError: err,
+				}
+			} else {
+				digests <- digestData{
+					digestValue:        digest,
+					containerName:      container.Name,
+					isServingContainer: len(rev.Spec.Containers) == 1 || len(container.Ports) != 0,
+				}
+			}
+			return nil
+		})
+	}
+	digestGrp.Wait()
+	close(digests)
+	for v := range digests {
+		if v.digestError != nil {
+			rev.Status.MarkContainerHealthyFalse(v1.ReasonContainerMissing,
+				v1.RevisionContainerMissingMessage(
+					v.image, v.digestError.Error()))
```

Comment:
Now I wonder if individual digest resolution errors should appear in each `ContainerStatus`

Reply:
I would think so 
because instead of giving single failure information for more than one images it would be better to give failure error respective to that

Reply:
let's leave it for a separate PR/issue